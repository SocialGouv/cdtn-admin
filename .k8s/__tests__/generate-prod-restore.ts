//

import { getEnvManifests } from "@socialgouv/kosko-charts/testing";
import { project } from "@socialgouv/kosko-charts/testing/fake/gitlab-ci.env";

jest.setTimeout(1000 * 60);
test("kosko generate --prod jobs/restore", async () => {
  expect(
    await getEnvManifests("prod", "jobs/restore", {
      ...project("cdtn-admin").dev,
    })
  ).toMatchSnapshot();
});
