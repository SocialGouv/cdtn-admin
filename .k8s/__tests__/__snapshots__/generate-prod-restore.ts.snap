// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kosko generate --prod jobs/restore/container when TRIGGER is set to PREPROD 1`] = `
"apiVersion: batch/v1
kind: Job
metadata:
  name: restore-container-prod-0123456
  namespace: cdtn-admin-secret
  labels:
    application: cdtn-admin
    owner: cdtn-admin
    team: cdtn-admin
    component: restore-prod
  annotations:
    kapp.k14s.io/update-strategy: always-replace
spec:
  backoffLimit: 0
  template:
    metadata:
      annotations:
        kapp.k14s.io/deploy-logs: for-new-or-existing
    spec:
      containers:
        - command:
            - sh
            - '-c'
            - >+


              [ ! -z $SOURCE_CONTAINER ] || (echo \\"No SOURCE_CONTAINER\\"; exit 1)

              [ ! -z $SOURCE_ACCOUNT_NAME ] || (echo \\"No SOURCE_ACCOUNT_NAME\\";
              exit 1)

              [ ! -z $SOURCE_ACCOUNT_KEY ] || (echo \\"No SOURCE_ACCOUNT_KEY\\";
              exit 1)

              [ ! -z $DESTINATION_CONTAINER ] || (echo \\"No
              DESTINATION_CONTAINER\\"; exit 1)

              [ ! -z $DESTINATION_ACCOUNT_NAME ] || (echo \\"No
              DESTINATION_ACCOUNT_NAME\\"; exit 1)

              [ ! -z $DESTINATION_ACCOUNT_KEY ] || (echo \\"No
              DESTINATION_ACCOUNT_KEY\\"; exit 1)



              echo \\"starting restore container $SOURCE_CONTAINER into
              $DESTINATION_CONTAINER\\"


              # copy container

              az storage copy -r   --source-account-name
              \\"$SOURCE_ACCOUNT_NAME\\"   --source-account-key
              \\"$SOURCE_ACCOUNT_KEY\\"   --source-container \\"$SOURCE_CONTAINER\\"  
              --account-name \\"$DESTINATION_ACCOUNT_NAME\\"   --account-key
              \\"$DESTINATION_ACCOUNT_KEY\\"   --destination-container
              \\"$DESTINATION_CONTAINER\\"


              # get original permission

              PUBLIC_ACCESS=\`az storage container show-permission  
              --account-name \\"$SOURCE_ACCOUNT_NAME\\"   --account-key
              \\"$SOURCE_ACCOUNT_KEY\\"   --name $SOURCE_CONTAINER | jq -r
              \\".publicAccess\\"\`


              # set original permission

              az storage container set-permission   --name
              $DESTINATION_CONTAINER   --account-name
              \\"$DESTINATION_ACCOUNT_NAME\\"   --account-key
              \\"$DESTINATION_ACCOUNT_KEY\\"   --public-access $PUBLIC_ACCESS

          env:
            - name: SOURCE_ACCOUNT_NAME
              valueFrom:
                secretKeyRef:
                  key: azurestorageaccountname
                  name: azure-cdtnadmindev-volume
            - name: SOURCE_ACCOUNT_KEY
              valueFrom:
                secretKeyRef:
                  key: azurestorageaccountkey
                  name: azure-cdtnadmindev-volume
            - name: DESTINATION_ACCOUNT_NAME
              valueFrom:
                secretKeyRef:
                  key: azurestorageaccountname
                  name: azure-cdtnadmindev-volume
            - name: DESTINATION_ACCOUNT_KEY
              valueFrom:
                secretKeyRef:
                  key: azurestorageaccountkey
                  name: azure-cdtnadmindev-volume
            - name: SOURCE_CONTAINER
              value: cdtn
            - name: DESTINATION_CONTAINER
              value: cdtn-preprod
          envFrom: []
          image: mcr.microsoft.com/azure-cli:2.15.1
          imagePullPolicy: IfNotPresent
          name: restore-db
          resources:
            limits:
              cpu: 300m
              memory: 512Mi
            requests:
              cpu: 50m
              memory: 64Mi
      restartPolicy: Never
  ttlSecondsAfterFinished: 86400
"
`;

exports[`kosko generate --prod jobs/restore/container when TRIGGER is set to PROD 1`] = `
"apiVersion: batch/v1
kind: Job
metadata:
  name: restore-container-prod-0123456
  namespace: cdtn-admin-secret
  labels:
    application: cdtn-admin
    owner: cdtn-admin
    team: cdtn-admin
    component: restore-prod
  annotations:
    kapp.k14s.io/update-strategy: always-replace
spec:
  backoffLimit: 0
  template:
    metadata:
      annotations:
        kapp.k14s.io/deploy-logs: for-new-or-existing
    spec:
      containers:
        - command:
            - sh
            - '-c'
            - >+


              [ ! -z $SOURCE_CONTAINER ] || (echo \\"No SOURCE_CONTAINER\\"; exit 1)

              [ ! -z $SOURCE_ACCOUNT_NAME ] || (echo \\"No SOURCE_ACCOUNT_NAME\\";
              exit 1)

              [ ! -z $SOURCE_ACCOUNT_KEY ] || (echo \\"No SOURCE_ACCOUNT_KEY\\";
              exit 1)

              [ ! -z $DESTINATION_CONTAINER ] || (echo \\"No
              DESTINATION_CONTAINER\\"; exit 1)

              [ ! -z $DESTINATION_ACCOUNT_NAME ] || (echo \\"No
              DESTINATION_ACCOUNT_NAME\\"; exit 1)

              [ ! -z $DESTINATION_ACCOUNT_KEY ] || (echo \\"No
              DESTINATION_ACCOUNT_KEY\\"; exit 1)



              echo \\"starting restore container $SOURCE_CONTAINER into
              $DESTINATION_CONTAINER\\"


              # copy container

              az storage copy -r   --source-account-name
              \\"$SOURCE_ACCOUNT_NAME\\"   --source-account-key
              \\"$SOURCE_ACCOUNT_KEY\\"   --source-container \\"$SOURCE_CONTAINER\\"  
              --account-name \\"$DESTINATION_ACCOUNT_NAME\\"   --account-key
              \\"$DESTINATION_ACCOUNT_KEY\\"   --destination-container
              \\"$DESTINATION_CONTAINER\\"


              # get original permission

              PUBLIC_ACCESS=\`az storage container show-permission  
              --account-name \\"$SOURCE_ACCOUNT_NAME\\"   --account-key
              \\"$SOURCE_ACCOUNT_KEY\\"   --name $SOURCE_CONTAINER | jq -r
              \\".publicAccess\\"\`


              # set original permission

              az storage container set-permission   --name
              $DESTINATION_CONTAINER   --account-name
              \\"$DESTINATION_ACCOUNT_NAME\\"   --account-key
              \\"$DESTINATION_ACCOUNT_KEY\\"   --public-access $PUBLIC_ACCESS

          env:
            - name: SOURCE_ACCOUNT_NAME
              valueFrom:
                secretKeyRef:
                  key: azurestorageaccountname
                  name: azure-cdtnadmindev-volume
            - name: SOURCE_ACCOUNT_KEY
              valueFrom:
                secretKeyRef:
                  key: azurestorageaccountkey
                  name: azure-cdtnadmindev-volume
            - name: DESTINATION_ACCOUNT_NAME
              valueFrom:
                secretKeyRef:
                  key: azurestorageaccountname
                  name: azure-cdtnadminprod-volume
            - name: DESTINATION_ACCOUNT_KEY
              valueFrom:
                secretKeyRef:
                  key: azurestorageaccountkey
                  name: azure-cdtnadminprod-volume
            - name: SOURCE_CONTAINER
              value: cdtn
            - name: DESTINATION_CONTAINER
              value: cdtn
          envFrom: []
          image: mcr.microsoft.com/azure-cli:2.15.1
          imagePullPolicy: IfNotPresent
          name: restore-db
          resources:
            limits:
              cpu: 300m
              memory: 512Mi
            requests:
              cpu: 50m
              memory: 64Mi
      restartPolicy: Never
  ttlSecondsAfterFinished: 86400
"
`;
